### PR TITLE
solution

### PR DESCRIPTION
**Pull request by mseshachalam**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+type ruleArea struct {
+	Name         string
+	ProvinceName string
+	CountryName  string
+}
+
+func (ra *ruleArea) String() string {
+	area := toInput(ra.Name) + "-" + toInput(ra.ProvinceName) + "-" + toInput(ra.CountryName)
+	for i, c := range area {
```
Comment (sivachandran) [inline main.go]: Can't we use `Trim` here?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+	return area
+}
+
+func subAreas(d *distributor) []*ruleArea {
+	var sas []*ruleArea
+
+	tcities := cities
+
+	for d != nil {
```
Comment (sivachandran) [inline main.go]: Instead of enumerating cities from leaf to root distributor if we do from root to leaf, will we not avoid processing too many cities?

**Comment by mseshachalam** [comment]
Added root to child area filtering

**Comment by mseshachalam** [comment]
Instead of string comparison of city data one by one, we could do build a tree of areas and look into.

**Pull request closed without merging**